### ISSUE TITLE
Beam search on BART seq2seq

### ISSUE THREAD

**Issue by ashleylew** [huggingface/transformers#11211]
## Environment info

- `transformers` version: 4.5.0.dev0
- Platform: Linux-3.10.0-1160.15.2.el7.x86_64-x86_64-with-glibc2.10
- Python version: 3.8.5
- PyTorch version (GPU?): 1.8.1 (False)
- Tensorflow version (GPU?): not installed (NA)
- Using GPU in script?: yes
- Using distributed or parallel set-up in script?: ?

### Who can help

@patrickvonplaten, @patil-suraj

## Information

Model I am using (Bert, XLNet ...): BART seq2seq

The problem arises when using:
* [x] the official example scripts: (give details below)
* [ ] my own modified scripts: (give details below)

The tasks I am working on is:
* [ ] an official GLUE/SQUaD task: (give the name)
* [x] my own task or dataset: (give details below)

## To reproduce

1.  I'm attempting to use beam search and have the model output the 10 best possible predictions for each test item.
2. I found the parameter `num_beams`, which I am using though it does not appear to work by itself. No error occurs, but only 1 output per test item is produced.
3. I thought I should use the parameter `num_return_sequences` as well, but it does not appear to be a possible argument for this model and I have not been able to find anything comparable. 

Here is my command:
```
python transformers/examples/seq2seq/run_translation.py \
    --model_name_or_path facebook/bart-base \
    --do_train \
    --do_predict \
    --source_lang en \
    --target_lang lf \
    --source_prefix "translate English to Logical Forms: " \
    --train_file folds/0_train.json \
    --test_file folds/0_val.json \
    --num_train_epochs=5 \
    --num_beams=10 \
    --output_dir ./test_results_beam \
    --per_device_train_batch_size=4 \
    --per_device_eval_batch_size=4 \
    --overwrite_output_dir \
    --predict_with_generate
```

## Expected behavior

Outputting the 10 best predictions per test item.

**Comment by patil-suraj** [comment]
Hi @ashleylew 

The `run_translation.py` uses the `Seq2SeqTrainer` which does not pass the `num_return_sequences` argument to `generate`, this is because if multiple sequences are returned then its not clear what sequence should be used to compute the metrics. 

you could generate test set predictions by using the `generate` method and passing the `num_return_sequences` argument. But if you want to do this using `Seq2SeqTrainer` then you would need to modify it.
